### PR TITLE
Improve the error handling for Follow Operations method on GCP

### PR DESCRIPTION
**Pull request by gabrielbussolo**
Signed-off-by: Gabriel Bussolo <gabriel@calyptia.com>

# Summary of this proposal

During the e2e test [i caught a mistake while evaluating errors coming back from GCP](https://github.com/calyptia/cloud-e2e/actions/runs/3096039417/jobs/5011119422#step:11:252), so this purpose improves the error handling for operations that is happening on deployment manager.

## Issue(s) number(s)

*Disclaimer: Please do not create a Pull Request without creating an issue first.*

## Checklist for submitter

- [ ] My PR has a related issue/bug number.
- [ ] My PR provides tests
  - [ ] Integration tests are added/passes
  - [ ] Unit tests are added/passes
  - [ ] End-to-end tests added/passes
  - [ ] Static code check added/passes
  - [ ] Linting on documentation added/passes
  - [ ] Does not affect code coverage stats
- [ ] My PR requires updating dependencies
- [ ] My PR has the documentation changes required.

## Checklist for reviewer

- [ ] The proposal fixes a bug/issue or implements a new feature that is well described.
- [ ] The proposal has sufficient test cases that covers the changes.
  - [ ] If changes an API, it does not break backwards compatibility (-1 major version).
- [ ] If integration is required, the proposal has integration tests.
- [ ] The proposal does not break coverage stats
- [ ] The proposal has the required documentation changes.

## Backport

- [ ] Backport to the latest stable release.

### CODE REVIEW

File: cmd/calyptia/delete_core_instance_gcp.go [hunk 1]
```diff
@@ -46,8 +46,8 @@ func newCmdDeleteCoreInstanceOnGCP(config *config, client gcp.Client) *cobra.Com
 			for {
 				operation, err := client.FollowOperations(ctx)
 
-				if err != nil || operation.Error != nil {
-					cmd.PrintErrf("an error occurred with the operation %s", operation.Name)
+				if err != nil {
+					cmd.PrintErrf("an error occurred with the operation: %v", err)
```
Comment (niedbalski) [inline cmd/calyptia/delete_core_instance_gcp.go:50]: with the operation %s: %v", operation.Name, err ?

File: gcp/client.go [hunk 2]
```diff
@@ -79,7 +79,11 @@ func New(ctx context.Context, projectName string, environment string, credential
 }
 
 func (c *DefaultClient) FollowOperations(ctx context.Context) (*deploymentmanager.Operation, error) {
-	return c.manager.Operations.Get(c.projectName, c.deploymentName).Context(ctx).Do()
+	operation, err := c.manager.Operations.Get(c.projectName, c.deploymentName).Context(ctx).Do()
+	if operation != nil && operation.Error != nil {
+		return operation, fmt.Errorf("operation error: %v", operation.Error.Errors)
```
Comment (niedbalski) [inline gcp/client.go:84]: operation %s error: %v" ? operation.Name

**Review by niedbalski — CHANGES_REQUESTED** [review]
minor nitpicks

**Comment by gabrielbussolo** [comment]
> minor nitpicks

i don't have a guarantee of a name if i cant have the operation so im printing the name with the operation error... take a look if its better now

**Pull request merged**